core/src/hir/lifetimes.rs

```diff
@@ -1,38 +1,36 @@
 //! Lifetime information for types.
 #![allow(dead_code)]
 
-use super::{IdentBuf, ImplicitLifetimeGenerator};
+use super::IdentBuf;
 use crate::ast;
-use smallvec::SmallVec;
+use smallvec::{smallvec, SmallVec};
+
+const EXPECTED_LIFETIMES: usize = 4;
```

Comment:
nit: LIFETIMES_SMALLVEC_INLINE_SIZE with docs? also it should probably be used in elision.rs too

Reply:
actually i'm not sure if that's the right name, i don't quite know what this is for